### ISSUE TITLE
Introduce appropriate public API for tuple conversions

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#11803]
Now that we have more than one tuple conversions, ```public bool IsTupleLiteral``` does not seem enough, and is possibly not accurate enough.

Confirm what we want as a public API here and implement

**Comment by gafter** [comment]
@DustinCampbell @CyrusNajmabadi We are considering pushing this out of the 2.0 milestone. Cyrus mentions that this might affect refactoring (expand and simplify), but suspects this is OK to push to a future release. Please let us know your position on this.

**Comment by CyrusNajmabadi** [comment]
As an example, my concern would be: what does hte IDE do now for certain refactoring for code like so:

```c#
(int, int) x = Foo();
...
public (byte, byte) Foo() { }
```

If we do some refactoring and complexify converts the above to ```(int, int) x = (byte, byte)Foo();``` and simplify doesn't fix it, that's an issue.

Complexify/Simplify covers a ton of cases.  I think we're going to have to do some good investigative testing here to see how well those cases work in the presence of tuples.

**Comment by VSadov** [comment]
The example given by Cyrus should work.

The current APIs would be insufficient in scenarios where you need to get to conversions of the individual elements. - For example if you want to splat a tuple into multiple locals, or something of that sort.

**Comment by jcouv** [comment]
Yes, I verified that the "rename" refactoring indeed works well in Cyrus's scenario. I didn't explore much around that, but Vlad makes a compelling point.

**Comment by CyrusNajmabadi** [comment]
Closing out as we haven't needed this.
